modules/openxr/openxr_api_extension.h

```diff
@@ -82,6 +82,23 @@ class OpenXRAPIExtension : public RefCounted {
 	void register_composition_layer_provider(OpenXRExtensionWrapperExtension *p_extension);
 	void unregister_composition_layer_provider(OpenXRExtensionWrapperExtension *p_extension);
 
+	void register_projection_views_extension(OpenXRExtensionWrapperExtension *p_extension);
+	void unregister_projection_views_extension(OpenXRExtensionWrapperExtension *p_extension);
+
+	double get_render_state_z_near();
+	double get_render_state_z_far();
```

Comment:
More thoughts on naming :-)

Internally, these values are on the `render_state` struct, but as a public API, I think folks would think of these as camera settings. Maybe it would be better as `get_camera_z_near()` and `get_camera_z_far()`?

Reply:
No this is important because render state specifically targets the values accessible from the render thread and have the state of the frame currently being processed by the render thread, while the main thread may be further along.

These functions also need to have a thread guard and give an error when called from anything but the render thread.

So I think its very important to keep this naming.

Reply:
I definitely agree about adding the thread guard! The spacewarp extension should already be calling those on the render thread.

I'd still personally prefer removing "render_state" from the function name, but I'd be OK with keeping the existing name, it's fine :-)